src/sentry/utils/metrics.py

```diff
@@ -58,6 +60,18 @@ def _add_global_tags(_all_threads: bool = False, **tags: TagValue) -> List[Tags]
     return stack
 
 
+def _filter_tags(tags: MutableTags) -> MutableTags:
+    """Removes unwanted tags from the tag mapping and returns a filtered one."""
+    discarded = frozenset(key for key in tags if key.endswith("_id") or key in BAD_TAGS)
+    if not discarded:
+        return tags
+
+    if settings.SENTRY_METRICS_WARN_BAD_TAGS:
+        logger = logging.getLogger("sentry.metrics")
+        logger.warning("discarded illegal metric tags: %s", sorted(discarded))
```

Comment:
Tbh I think we should just throw an exception here. It'll help us catch everything in ci, and make it harder for people to mess up